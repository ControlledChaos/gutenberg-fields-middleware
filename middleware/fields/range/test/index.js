/**
 * External dependencies.
 */
import { render } from 'enzyme';

/**
 * Internal dependencies.
 */
import { range } from '../../';

describe( 'RangeControl', () => {
	const config = {
		type: 'range',
		label: 'Select Range',
		help: 'This is your range control',
		beforeIcon: 'editor-textcolor',
		afterIcon: 'editor-textcolor',
		placement: 'inspector',
		min: 0,
		max: 20,
	};

	const wrapper = render( range( '', config, 'radio' ) );

	test( 'should render a range Control', () => {
		expect( wrapper ).toMatchSnapshot();
	} );

} );
