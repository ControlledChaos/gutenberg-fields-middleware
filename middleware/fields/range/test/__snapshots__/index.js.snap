// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeControl should render a range Control 1`] = `
<div
  class="components-base-control components-range-control"
>
  <div
    class="components-base-control__field"
  >
    <label
      class="components-base-control__label"
      for="inspector-range-control-0"
    >
      Select Range
    </label>
    <svg
      aria-hidden="true"
      class="dashicon dashicons-editor-textcolor"
      focusable="false"
      height="20"
      role="img"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.23 15h1.9L11 4H9L5 15h1.88l1.07-3h4.18zm-1.53-4.54H8.51L10 5.6z"
      />
    </svg>
    <input
      aria-describedby="inspector-range-control-0__help"
      class="components-range-control__slider"
      id="inspector-range-control-0"
      max="20"
      min="0"
      placement="inspector"
      type="range"
      value=""
    />
    <svg
      aria-hidden="true"
      class="dashicon dashicons-editor-textcolor"
      focusable="false"
      height="20"
      role="img"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.23 15h1.9L11 4H9L5 15h1.88l1.07-3h4.18zm-1.53-4.54H8.51L10 5.6z"
      />
    </svg>
    <input
      aria-label="Select Range"
      class="components-range-control__number"
      max="20"
      min="0"
      placement="inspector"
      type="number"
      value=""
    />
  </div>
  <p
    class="components-base-control__help"
    id="inspector-range-control-0__help"
  >
    This is your range control
  </p>
</div>
`;
